smithy-model/src/test/java/software/amazon/smithy/model/transform/CreateDedicatedInputAndOutputTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.model.transform;
+
+import java.util.function.Function;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.shapes.ModelSerializer;
+
+public class CreateDedicatedInputAndOutputTest {
+    private void compareTransform(String prefix, Function<Model, Model> transform) {
+        Model before = Model.assembler()
+                .addImport(getClass().getResource("dedicated-input-output/" + prefix + "-before.smithy"))
+                .assemble()
+                .unwrap();
+        Model actualResult = transform.apply(before);
+        Model expectedResult = Model.assembler()
+                .addImport(getClass().getResource("dedicated-input-output/" + prefix + "-after.smithy"))
+                .assemble()
+                .unwrap();
+
+        Node actualNode = ModelSerializer.builder().build().serialize(actualResult);
+        Node expectedNode = ModelSerializer.builder().build().serialize(expectedResult);
+
+        Node.assertEquals(actualNode, expectedNode);
```

Comment:
Since the original id trait isn't serialized, you're not asserting in these tests that it's applied at all

Reply:
Very good catch. I actually had a couple misses because of this.